LuckyDragon/app/src/main/java/com/example/luckydragon/User.java

```diff
@@ -71,11 +71,11 @@ public void fetchData() {
                         // create new user with empty info
                         save();
                     } else {
-                        email = userData.get("email") != null ? String.format("%s", userData.get("email")) : "";
-                        name = userData.get("name") != null ? String.format("%s", userData.get("name")) : "";
-                        phoneNumber = userData.get("phoneNumber") != null ? String.format("%s", userData.get("phoneNumber")) : "";
-                        notifications = userData.get("notifications") != null
-                                && userData.get("notifications").toString().equals("true");
+                        name = userData.get("name") != null ? Objects.requireNonNull(userData.get("name")).toString() : null;
+                        email = userData.get("email") != null ? Objects.requireNonNull(userData.get("email")).toString() : null;
+                        phoneNumber = userData.get("phoneNumber") != null ? Objects.requireNonNull(userData.get("phoneNumber")).toString() : null;
+                        assert(name != null);
+
```

Comment:
Objects requireNonNull and ternary operator are redundant

Reply:
Value will be "null" (String) instead of null without ternary operator and Android Studio gives warning if not for Objects requireNonNull

Reply:
I think that is because the db stored "null" as the name. If we fix the db to never save the attribute if it is null, this can be removed.

eg. Objects.requireNonNull(userData.get("email")).toString();

Reply:
After reading your other comments, I think this still applies btw. map.get() will return a null pointer if it doesn't exist or will return the null pointer if we store into the db null (not the string).

Ie. this can be simplified to `Objects.requireNonNull()`

Reply:
The db can save the attribute to null so I am keeping it